src/lua/datetime.lua

```diff
@@ -936,7 +940,20 @@ local function datetime_parse_from(str, obj)
     -- Override timezone, if it was not specified in a parsed
     -- string.
     if date.tz == '' and date.tzoffset == 0 then
-        datetime_set(date, { tzoffset = tzoffset, tz = tzname })
+        local offset = nil
+
+        if tzoffset ~= nil then
+            offset = get_timezone(tzoffset, 'tzoffset')
+            check_range(offset, -720, 840, 'tzoffset')
+        end
+
+        if tzname ~= nil then
+            offset, date.tzindex = parse_tzname(date.epoch, tzname)
+        end
+
+        if offset ~= nil then
+            time_localize(date, offset)
+        end
```

Comment:
This changes are already covered by the existing tests.
The change is needed since before the function used `datetime_set()` call for specifying a `tzoffset` and `tzname`.

But after the patch, `datetime_set` with a provided timezone not only changes the timezone but affect the hour/minute values. Though we only need to roughly set the timezone since hours/minutes are already properly parsed and calculated.

Thus I've copied the code from the `datetime_set` implementation here. It's already covered in the variety of tests on parsing.

```diff
@@ -936,7 +942,20 @@ local function datetime_parse_from(str, obj)
     -- Override timezone, if it was not specified in a parsed
     -- string.
     if date.tz == '' and date.tzoffset == 0 then
-        datetime_set(date, { tzoffset = tzoffset, tz = tzname })
+        local offset = nil
+
+        if tzoffset ~= nil then
+            offset = get_timezone(tzoffset, 'tzoffset')
+            check_range(offset, -720, 840, 'tzoffset')
+        end
+
+        if tzname ~= nil then
+            offset, date.tzindex = parse_tzname(date.epoch, tzname)
+        end
+
+        if offset ~= nil then
+            time_localize(date, offset)
+        end
     end
```